sormas-backend/src/main/java/de/symeda/sormas/backend/user/CurrentUserCache.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2022 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+package de.symeda.sormas.backend.user;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Create a mapping between the username provided by the session context and the actual DB entry to save round trips.
+ */
+public final class CurrentUserCache {
+
+	private static CurrentUserCache instance;
+
+	ConcurrentHashMap<String, CurrentUser> cache;
+
+	private CurrentUserCache() {
+		cache = new ConcurrentHashMap<>();
+	}
+
+	public static CurrentUserCache getInstance() {
+		if (instance == null) {
+			instance = new CurrentUserCache();
+		}
+		return instance;
+	}
+
+	public void flush() {
+		cache.clear();
+	}
+
+	public void put(String name, CurrentUser user) {
+		cache.put(name, user);
```

Comment:
When using Keycloak the username we receive from Keycloak is always with lowercase letters, however the username in the DB is still stored in PascalCase format, which might introduce some conflicts for example when removing a user from the cache.
For example, the callerPrincipal from Keycloak will be "natuser", but when you update the user, you want to remove it from cache, so it will try to remove "NatUser" which it won't find.
My suggestion is to use `AuthProvider.isUsernameCaseSensitive` which if it's true, then convert all usernames to "lowercase" before `putting`, `getting`, `removing` them from the map

Reply:
Hi Alex I filed https://github.com/hzi-braunschweig/SORMAS-Security/issues/39 to discuss this further. Currently, I don't want to merge the PR as I do not want to introduce a vulnerability here. I'll leave it as draft for now till the concerns I have are gone :smiley:

Reply:
Resolved by #7907